src/main/scala/com/tuplejump/snackfs/api/model/FileStatusCommand.scala

```diff
@@ -18,32 +18,37 @@
  */
 package com.tuplejump.snackfs.api.model
 
-import scala.concurrent.duration.FiniteDuration
-import scala.util.{Failure, Success, Try}
+import java.io.FileNotFoundException
+
 import scala.concurrent.Await
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+import org.apache.hadoop.fs.FileStatus
+import org.apache.hadoop.fs.Path
+
+import com.tuplejump.snackfs.SnackFS
+import com.tuplejump.snackfs.api.partial.Command
+import com.tuplejump.snackfs.cassandra.partial.FileSystemStore
 import com.tuplejump.snackfs.fs.model.INode
-import java.io.FileNotFoundException
+import com.tuplejump.snackfs.util.LogConfiguration
 import com.twitter.logging.Logger
-import org.apache.hadoop.fs.{FileStatus, Path}
-import com.tuplejump.snackfs.cassandra.partial.FileSystemStore
-import com.tuplejump.snackfs.api.partial.Command
 
 object FileStatusCommand extends Command {
   private lazy val log = Logger.get(getClass)
 
-  def apply(store: FileSystemStore,
-            filePath: Path,
-            atMost: FiniteDuration): FileStatus = {
-
-    log.debug("getting status for %s", filePath)
+  def apply(fs: SnackFS, store: FileSystemStore, filePath: Path, atMost: FiniteDuration): FileStatus = {
     val maybeFile = Try(Await.result(store.retrieveINode(filePath), atMost))
-
     maybeFile match {
-      case Success(file: INode) => SnackFileStatus(file, filePath)
+      case Success(file: INode) =>
+        val absolutePath = fs.resolvePath(new Path(fs.getWorkingDirectory, filePath))
```

Comment:
Is this really needed? Is there any other way to do a resolvePath here?

This breaks the existing API and contracts for all the Commands.